### PR TITLE
Tree view

### PR DESCRIPTION
**Pull request by madhur-tandon**
Closes https://github.com/jupyterlab-contrib/search-replace/issues/21 and https://github.com/jupyterlab-contrib/search-replace/issues/22

### CODE REVIEW

File: src/searchReplace.tsx [hunk 1]
```diff
@@ -52,10 +58,56 @@ export class SearchReplaceModel extends VDomModel {
...
+    start: number;
+    end: number;
+    match: string;
+    line_number: number;
+    absolute_offset: number;
+  }[];
+}
+
+function createTreeView(results: IResults[]): JSX.Element {
+  const items = results.map(file => {
```
Comment (fcollonval) [inline src/searchReplace.tsx:82]: We need to sort the results in a deterministic way. A good sorting will be to sort per path level and within a directory per filename.

File: src/searchReplace.tsx [hunk 2]
```diff
@@ -52,10 +58,56 @@ export class SearchReplaceModel extends VDomModel {
...
+  path: string;
+  matches: {
+    line: string;
+    start: number;
+    end: number;
+    match: string;
+    line_number: number;
+    absolute_offset: number;
+  }[];
+}
```
Comment (fcollonval) [inline src/searchReplace.tsx:79]: Could you add doc string for this?

> In JS a doc string is defined in block like this:

```js
/**
 * My documentation
 */
```

> The important thing is the `/**` to distinguish it from standard block comments.

**Review by fcollonval — CHANGES_REQUESTED** [review]
I pushed some update to improve the look of the UI. Could you address the two following points?

File: src/searchReplace.tsx (hunk 2 above)
Comment (madhur-tandon) [reply src/searchReplace.tsx:91]: done

File: ui-tests/tests/search.spec.ts [hunk 3]
```diff
@@ -1,25 +1,66 @@
...
+    `${tmpPath}/${fileName}`
+  );
+});
+
+test.afterAll(async ({ baseURL, tmpPath }) => {
+  const contents = galata.newContentsHelper(baseURL);
+  await contents.deleteDirectory(tmpPath);
+});
+
+test('should get 5 matches', async ({ page }) => {
```
Comment (fcollonval) [inline ui-tests/tests/search.spec.ts:24]: As you included #22 in this one, could you add a test for opening a file from a click.

**Review by fcollonval — CHANGES_REQUESTED** [review]

File: ui-tests/tests/search.spec.ts (hunk 3 above)
Comment (madhur-tandon) [reply ui-tests/tests/search.spec.ts:24]: done

**Comment by madhur-tandon** [comment]
@fcollonval ready to merge I guess

**Review by fcollonval — APPROVED** [review]
Thanks @madhur-tandon

**Pull request merged**